### ISSUE TITLE
Make touchswipe plugin CommonJS compatible

### ISSUE THREAD

**Issue by igorsechyn** [mattbryson/TouchSwipe-Jquery-Plugin#223]
I was trying to use this plugin with a brwoserify setup, where each module is a CommonJS module. Normally to include an external jquery plugin, it should be sufficient to require it somewhere. Unfortunately it does not work with touch swipe plugin, since it relies on a global jQuery object, which is not present when using the noconflict mode. This could be easily fixed by enhancing the factory call as follows:
```javascript
function (factory) {
   if (typeof define === 'function' && define.amd && define.amd.jQuery) {
        // AMD. Register as anonymous module.
        define(['jquery'], factory);
    }
// CommonJS Module
		if (typeof module !== 'undefined' && module.exports) {
			  factory(require("jquery"));
		}
		else {
        // Browser globals.
        factory(jQuery);
    }
}(function ($) {

``` 
otherwise i have to expose the jquery object as global jQuery

**Comment by MartijnR** [comment]
+1

**Comment by mattbryson** [comment]
fixed in 1.6.14
